### PR TITLE
style: improve modules/default.nix code, rename helper scripts

### PR DESCRIPTION
**Pull request by diniamo**

### CODE REVIEW

File: modules/default.nix [hunk 1]
```diff
@@ -100,29 +98,40 @@ inputs: {
...
   # generate a wrapped Neovim package.
   neovim-wrapped = inputs.mnw.lib.wrap pkgs {
     neovim = vimOptions.package;
     plugins = concatLists [builtStartPlugins builtOptPlugins];
     appName = "nvf";
     extraBinPath = vimOptions.extraPackages;
-    initLua = vimOptions.builtLuaConfigRC;
-    luaFiles = vimOptions.extraLuaFiles;
+    # initLua = vimOptions.builtLuaConfigRC;
+    # luaFiles = vimOptions.extraLuaFiles;
```
Comment (NotAShelf) [inline modules/default.nix:119]: Is there a reason those two haave been left out?

**Review by NotAShelf — DISMISSED** [review]

File: modules/default.nix (hunk 1 above)
Comment (diniamo) [reply modules/default.nix:119]: Forgot to remove them

**Comment by Gerg-L** [comment]
Add
```
 vimCommandCheckHook 
vimGenDocHook
neovimRequireCheckHook
```
To nativeBuildInputs
And then set `passthru.vimPlugin = true;`

File: modules/default.nix [hunk 2]
```diff
@@ -100,29 +100,38 @@ inputs: {
   extraLuaPackages = ps: map (x: ps.${x}) vimOptions.luaPackages;
   extraPython3Packages = ps: map (x: ps.${x}) vimOptions.python3Packages;
 
+  luaConfig =
+    if vimOptions.byteCompileLua
+    then pkgs.runCommandLocal "init.lua" {text = vimOptions.builtLuaConfigRC;} "${pkgs.luajit}/bin/luajit -bd -- - $out <<< \"$text\""
```
Comment (Gerg-L) [inline modules/default.nix:105]: `pkgs.luajit` should be the unwrapped neovim's `.lua` attribute

File: modules/default.nix [hunk 3]
```diff
@@ -100,29 +100,38 @@ inputs: {
...
     inherit extraLuaPackages extraPython3Packages;
   };
 
   # Additional helper scripts for printing and displaying nvf configuration
   # in your commandline.
-  printConfig = pkgs.writers.writeDashBin "print-nvf-config" ''
+  printConfig = pkgs.writers.writeDashBin "nvf-print-config" ''
     cat << EOF
       ${vimOptions.builtLuaConfigRC}
     EOF
```
Comment (Gerg-L) [inline modules/default.nix:131]: wtf is this

Comment (diniamo) [reply modules/default.nix:131]: that was raf

File: modules/default.nix (hunk 2 above)
Comment (diniamo) [reply modules/default.nix:105]: What does that mean

Comment (diniamo) [reply modules/default.nix:105]: Oh

Comment (diniamo) [reply modules/default.nix:105]: The lua package doesn't provide luajit though

Comment (Gerg-L) [reply modules/default.nix:105]:
```nix
$ nix build .#neovim-unwrapped.lua
$ ls result/bin
lua  luajit  luajit-2.1.1713773202
```

Comment (Gerg-L) [reply modules/default.nix:105]:
```bash
$ nix build github:nix-community/neovim-nightly-overlay#neovim.lua
$ ls result/bin
lua  luajit  luajit-2.1.1713773202
```

File: modules/default.nix (hunk 3 above)
Comment (diniamo) [reply modules/default.nix:135]: @Gerg-L do you think cat-ing the path returned by the other script is better? That's what I suggested as well.

Comment (Gerg-L) [reply modules/default.nix:135]: Make a variable which is the writeText then use it in the echo and cat script

**Review by NotAShelf — APPROVED** [review]

**Comment by Gerg-L** [comment]
@NotAShelf @diniamo this broke stuff

**Comment by NotAShelf** [comment]
how

**Pull request merged**